### ISSUE TITLE
There's upstream dependancy conflict regarding react while installing dependencies in the frontend

### ISSUE THREAD

**Issue by DevBlocker** [yenxuan1381/anti-counterfeit-product-identification-system-using-blockchain#3]
Cannot install the dependencies. Using --force or --legacy-peer-deps creates problems.

npm ERR! code ERESOLVE
npm ERR! ERESOLVE could not resolve
npm ERR!
npm ERR! While resolving: @material-ui/core@4.12.4
npm ERR! Found: react@18.2.0
npm ERR! node_modules/react
npm ERR!   react@"^18.2.0" from the root project
npm ERR!   peer react@">=16.8.0" from @emotion/react@11.10.6
npm ERR!   node_modules/@emotion/react
npm ERR!     @emotion/react@"^11.10.6" from the root project
npm ERR!     peer @emotion/react@"^11.0.0-rc.0" from @emotion/styled@11.10.6
npm ERR!     node_modules/@emotion/styled
npm ERR!       @emotion/styled@"^11.10.6" from the root project
npm ERR!       5 more (@mui/lab, @mui/material, @mui/styled-engine, ...)
npm ERR!     5 more (@mui/lab, @mui/material, @mui/styled-engine, ...)
npm ERR!   28 more (@emotion/styled, ...)
npm ERR!
npm ERR! Could not resolve dependency:
npm ERR! peer react@"^16.8.0 || ^17.0.0" from @material-ui/core@4.12.4
npm ERR! node_modules/@material-ui/core
npm ERR!   @material-ui/core@"^4.12.4" from the root project
npm ERR!   peer @material-ui/core@"^4.12.1" from @material-ui/lab@4.0.0-alpha.61
npm ERR!   node_modules/@material-ui/lab
npm ERR!     @material-ui/lab@"^4.0.0-alpha.61" from the root project
npm ERR!
npm ERR! Conflicting peer dependency: react@17.0.2
npm ERR! node_modules/react
npm ERR!   peer react@"^16.8.0 || ^17.0.0" from @material-ui/core@4.12.4
npm ERR!   node_modules/@material-ui/core
npm ERR!     @material-ui/core@"^4.12.4" from the root project
npm ERR!     peer @material-ui/core@"^4.12.1" from @material-ui/lab@4.0.0-alpha.61
npm ERR!     node_modules/@material-ui/lab
npm ERR!       @material-ui/lab@"^4.0.0-alpha.61" from the root project
npm ERR!
npm ERR! Fix the upstream dependency conflict, or retry
npm ERR! this command with --force or --legacy-peer-deps
npm ERR! to accept an incorrect (and potentially broken) dependency resolution.

provided only the conflict part ^.

**Comment by abdulfaizan11** [comment]
hey! did you solve that error  , I need a help

**Comment by DevBlocker** [comment]
No, it is irresolvable as ERESOLVE says. You can use --legacy-peer-deps while installing the dependencies, to bring it to a somewhat broken but workable state.
